### DISCUSSION TITLE
Error code ELIFECYCLE while installing Superset behind proxy

### DISCUSSION THREAD

**Discussion by pm19035** [apache/superset#18735] [General]
I am trying to install Superset for building custom visualization using the guide at: https://superset.apache.org/docs/installation/building-custom-viz-plugins/

Please note that I am attempting it on Ubuntu 20.04 VMs behind a proxy. I am able to install Superset properly in python venv using the guide available at https://superset.apache.org/docs/installation/installing-superset-from-scratch/

However, for tinkering with custom visualizations, I need to go the  npm run dev-server way which is causing me issues as it gives me "npm ERR! code ELIFECYCLE
npm ERR! errno 2".

#### How to reproduce the bug

cd superset/superset-frontend
npm install
npm run dev-server

### Expected results

Successful installation of Apache Superset

### Actual results

getting stuck at the npm run dev-server command while the docker-compose up is merrily running its loop.

10% building 0/1 entries 0/0 dependencies 0/0 modules[webpack-cli] /home/rrscw/superset/superset-frontend/node_module s/webpack-dev-server/lib/servers/WebsocketServer.js:10
static heartbeatInterval = 1000;
^

SyntaxError: Unexpected token =
at new Script (vm.js:83:7)
at NativeCompileCache._moduleCompile (/home/rrscw/superset/superset-frontend/node_modules/v8-compile-cache/v8-com pile-cache.js:240:18)
at Module._compile (/home/rrscw/superset/superset-frontend/node_modules/v8-compile-cache/v8-compile-cache.js:184: 36)
at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
at Module.load (internal/modules/cjs/loader.js:653:32)
at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
at Function.Module._load (internal/modules/cjs/loader.js:585:3)
at Module.require (internal/modules/cjs/loader.js:692:17)
at require (/home/rrscw/superset/superset-frontend/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
at Server.getServerTransport (/home/rrscw/superset/superset-frontend/node_modules/webpack-dev-server/lib/Server.j s:1009:28)
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! [superset@0.0.0-dev](mailto:superset@0.0.0-dev) dev-server: cross-env NODE_ENV=development BABEL_ENV=development node --max_old_space_si ze=4096 ./node_modules/webpack-dev-server/bin/webpack-dev-server.js --mode=development
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the [superset@0.0.0-dev](mailto:superset@0.0.0-dev) dev-server script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR! /root/.npm/_logs/2022-02-13T06_48_45_404Z-debug.log
root@rrscw:/home/rrscw/superset/superset-frontend# node /root/.npm/_logs/2022-02-13T06_48_45_404Z-debug.log

The log file says:

0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'dev-server' ]
2 info using npm@6.14.4
3 info using node@v10.19.0
4 verbose run-script [ 'predev-server', 'dev-server', 'postdev-server' ]
5 info lifecycle [superset@0.0.0-dev](mailto:superset@0.0.0-dev)predev-server: [superset@0.0.0-dev](mailto:superset@0.0.0-dev)
6 info lifecycle [superset@0.0.0-dev](mailto:superset@0.0.0-dev)dev-server: [superset@0.0.0-dev](mailto:superset@0.0.0-dev)
7 verbose lifecycle [superset@0.0.0-dev](mailto:superset@0.0.0-dev)dev-server: unsafe-perm in lifecycle true
8 verbose lifecycle [superset@0.0.0-dev](mailto:superset@0.0.0-dev)dev-server: PATH: /usr/share/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/rrscw/superset/superset-frontend/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/g>
9 verbose lifecycle [superset@0.0.0-dev](mailto:superset@0.0.0-dev)dev-server: CWD: /home/rrscw/superset/superset-frontend
10 silly lifecycle [superset@0.0.0-dev](mailto:superset@0.0.0-dev)dev-server: Args: [ '-c',
10 silly lifecycle 'cross-env NODE_ENV=development BABEL_ENV=development node --max_old_space_size=4096 ./node_modules/webpack-dev-server/bin/webpack-dev-server.js --mode=development' ]
11 silly lifecycle [superset@0.0.0-dev](mailto:superset@0.0.0-dev)dev-server: Returned: code: 2 signal: null
12 info lifecycle [superset@0.0.0-dev](mailto:superset@0.0.0-dev)dev-server: Failed to exec dev-server script
13 verbose stack Error: [superset@0.0.0-dev](mailto:superset@0.0.0-dev) dev-server: cross-env NODE_ENV=development BABEL_ENV=development node --max_old_space_size=4096 ./node_modules/webpack-dev-server/bin/webpack-dev-server.js --mode=development
13 verbose stack Exit status 2
13 verbose stack at EventEmitter. (/usr/share/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack at EventEmitter.emit (events.js:198:13)
13 verbose stack at ChildProcess. (/usr/share/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack at ChildProcess.emit (events.js:198:13)
13 verbose stack at maybeClose (internal/child_process.js:982:16)
13 verbose stack at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
14 verbose pkgid [superset@0.0.0-dev](mailto:superset@0.0.0-dev)
15 verbose cwd /home/rrscw/superset/superset-frontend
16 verbose Linux 5.4.0-99-generic
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "dev-server"
18 verbose node v10.19.0
19 verbose npm v6.14.4
20 error code ELIFECYCLE
21 error errno 2
22 error [superset@0.0.0-dev](mailto:superset@0.0.0-dev) dev-server: cross-env NODE_ENV=development BABEL_ENV=development node --max_old_space_size=4096 ./node_modules/webpack-dev-server/bin/webpack-dev-server.js --mode=development
22 error Exit status 2
23 error Failed at the [superset@0.0.0-dev](mailto:superset@0.0.0-dev) dev-server script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 2, true ]

I have no clue what is going on at this point.

### Environment

(please complete the following information):

OS: Ubuntu Server 20.04
Python on VM: 3.8.10
node on VM: 16.9.1
npm on VM: 8.1.2

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ ] I have reproduced the issue with at least the latest released version of superset.
- [ ] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by zhaoyongjie** [comment]
@pm19035 Please double-check `node` and `npm` versions for installation. It's the incompatible version from your log.

>>
>>16 verbose Linux 5.4.0-99-generic
>>17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "dev-server"
>>18 verbose node v10.19.0
>>19 verbose npm v6.14.4
